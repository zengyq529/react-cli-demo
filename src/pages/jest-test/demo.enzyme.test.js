import React from 'react'

import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App,{DemoComponent} from './enzymedemo';

Enzyme.configure({ adapter: new Adapter() });



describe('enzyme test', () => {
  test('enzyme', () => {
    const wrapper = shallow(<App />) // 浅渲染 不渲染子组件 适合单元测试
    console.log(wrapper.find('#app'));
    console.log(wrapper.find('.class-name').prop('title')) //少用 classname 变得可能性很大。
    console.log(wrapper.find('[data-test="container"]')) // 这种解耦比较好。

    // 安装 jest-enzyme 提供了更多匹配器 , 在jest.config.js 添加扩展配置 "setupFilesAfterEnv": ['./node_modules/jest-enzyme/lib/index.js'],
    expect(wrapper.find('[data-test="container"]')).toExist();

  })

  test('enzyme2', () => {
    const wrapper = mount(<App />) //全渲染 适合集成测试
    expect(wrapper).toMatchSnapshot() //如果对展示敏感的测试可以使用快照
  });

  test('enzyme 受控组件 模拟用户行为 验证受控组件能力', () => {
    let value = '123';
    const onChange = (v) => (value = v);
    const wrapper = mount(<App value={value} onChange={onChange} />)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('input').simulate('change', {
      target: { value: "456" }
    })
    expect(value).toBe('456');
    // wrapper.state('key')直接获取state的值
    // wrapper.setState()//
    // expect(fn).toHaveBeenCalled();
    // expect(fn).toHaveBeenLastCalledWith('值') //最后一次被调用的参数。
  });

  test('enzyme 快照，希望样式不被改变', () => {
    const wrapper = shallow(<DemoComponent />)
    expect(wrapper).toMatchSnapshot(); //通过构建快照，确认渲染结果相同。
  });

  test('enzyme 测试封装', () => {
    const wrapper = shallow(<DemoComponent />)
    expect(wrapper).toMatchSnapshot(); //通过构建快照，确认渲染结果相同。
  });

})
