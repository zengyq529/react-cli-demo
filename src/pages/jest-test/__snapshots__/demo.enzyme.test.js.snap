// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme test enzyme 受控组件 模拟用户行为 验证受控组件能力 1`] = `
<App
  onChange={[Function]}
  value="123"
>
  <div
    data-test="container"
    id="app"
    onClick={[Function]}
  >
    <div
      className="class-name"
      title="hhh"
    />
    <input
      onChange={[Function]}
      value="123"
    />
    0
  </div>
</App>
`;

exports[`enzyme test enzyme 快照，希望样式不被改变 1`] = `ShallowWrapper {}`;

exports[`enzyme test enzyme 测试封装 1`] = `ShallowWrapper {}`;

exports[`enzyme test enzyme2 1`] = `
<App>
  <div
    data-test="container"
    id="app"
    onClick={[Function]}
  >
    <div
      className="class-name"
      title="hhh"
    />
    0
  </div>
</App>
`;
